phidgets_motors/src/motors_ros_i.cpp

```diff
@@ -101,15 +101,18 @@ MotorsRosI::MotorsRosI(const rclcpp::NodeOptions& options)
             snprintf(pubtopic, sizeof(pubtopic), "motor_duty_cycle%02d", i);
             motor_vals_[i].duty_cycle_pub =
                 this->create_publisher<std_msgs::msg::Float64>(pubtopic, 1);
-
-            char backemftopic[] = "motor_back_emf00";
-            snprintf(backemftopic, sizeof(backemftopic), "motor_back_emf%02d",
-                     i);
-            motor_vals_[i].back_emf_pub =
-                this->create_publisher<std_msgs::msg::Float64>(backemftopic, 1);
-
             motor_vals_[i].last_duty_cycle_val = motors_->getDutyCycle(i);
-            motor_vals_[i].last_back_emf_val = motors_->getBackEMF(i);
+
+            if (motors_->backEMFSensingSupported(i))
```

Comment:
Honestly, I wouldn't bother with this.  It's somewhat subjective, but I'd rather just unconditionally create the topic.  If the underlying device doesn't support it, then we will never get a callback and never publish anything on it, which is fine.

I think you'll still want to keep the `backEMFSensingSupported` method so you can print an `RCLCPP_INFO` about whether it is supported or not.